### PR TITLE
[clang-linker-wrapper] Add error handling for missing linker path

### PR DESCRIPTION
**Pull request by asudarsa**
In clang-linker-wrapper, we do not explicitly check if --linker-path is not provided.
This PR adds a check to capture this.

Thanks

### CODE REVIEW

File: clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp [hunk 1]
```diff
@@ -411,6 +411,8 @@ Error runLinker(ArrayRef<StringRef> Files, const ArgList &Args) {
   // Render the linker arguments and add the newly created image. We add it
   // after the output file to ensure it is linked with the correct libraries.
   StringRef LinkerPath = Args.getLastArgValue(OPT_linker_path_EQ);
+  if (LinkerPath.empty())
+    return createStringError("Host linker is not available");
```
Comment (sarnex) [inline clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:415]: Should this go upstream?

File: clang/test/Driver/linker-wrapper-sycl-win.cpp [hunk 2]
```diff
@@ -112,3 +112,7 @@
 // CHK-CMDS-AOT-AMD-NEXT: offload-wrapper: input: [[BUNDLEROUT]], output: [[WRAPPEROUT:.*]].bc
 // CHK-CMDS-AOT-AMD-NEXT: "{{.*}}clang.exe"{{.*}} -c -o [[LLCOUT:.*]].o [[WRAPPEROUT]].bc
 // CHK-CMDS-AOT-AMD-NEXT: "{{.*}}ld" -- HOST_LINKER_FLAGS -dynamic-linker HOST_DYN_LIB -o a.out [[LLCOUT]].o HOST_LIB_PATH HOST_STAT_LIB {{.*}}.o
+
+// Error handling when --linker-path is not provided for clang-linker-wrapper
+// RUN: not clang-linker-wrapper 2>&1 | FileCheck --check-prefix=LINKER-PATH-NOT-PROVIDED %s
+// LINKER-PATH-NOT-PROVIDED: Host linker is not available
```
Comment (srividya-sundaram) [inline clang/test/Driver/linker-wrapper-sycl-win.cpp:118]: Per [LLVM Coding standards for error messages](https://llvm.org/docs/CodingStandards.html#error-and-warning-messages), the first sentence of the error message starts with a _lower-case letter_.

File: clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp (hunk 1 above)
Comment (asudarsa) [reply clang/tools/clang-linker-wrapper/ClangLinkerWrapper.cpp:415]: Yes. Happening in parallel.

https://github.com/llvm/llvm-project/pull/113613#issuecomment-2436215580

tests are different, but expect to converge after pulldown.

Thanks

**Review by sarnex — APPROVED** [review]
lgtm assuming the code will be the exact same upstream so we don't have a complicated merge during pulldown

File: clang/test/Driver/linker-wrapper-sycl-win.cpp (hunk 2 above)
Comment (asudarsa) [reply clang/test/Driver/linker-wrapper-sycl-win.cpp]: Thanks. addressed in next commit.

**Review by srividya-sundaram — APPROVED** [review]

**Comment by maarquitos14** [comment]
@intel/llvm-gatekeepers I think this is ready to merge.

**Pull request merged**